src/core/diags/factory_diags.cpp

```diff
@@ -543,6 +543,39 @@ void Diags::TransmitDone(Error aError)
 
 #endif // OPENTHREAD_RADIO
 
+Error Diags::ProcessGpio(uint8_t aArgsLength, char *aArgs[], char *aOutput, size_t aOutputMaxLen)
+{
+    Error    error = kErrorNone;
+    long     value;
+    uint32_t gpio;
+    bool     level;
+
+    if ((aArgsLength == 2) && (strcmp(aArgs[0], "get") == 0))
+    {
+        SuccessOrExit(error = ParseLong(aArgs[1], value));
+        gpio = static_cast<uint32_t>(value);
+        SuccessOrExit(error = otPlatDiagGpioGet(gpio, &level));
+        snprintf(aOutput, aOutputMaxLen, "%d\r\n", level);
+    }
+    else if ((aArgsLength == 3) && (strcmp(aArgs[0], "set") == 0))
+    {
+        SuccessOrExit(error = ParseLong(aArgs[1], value));
+        gpio = static_cast<uint32_t>(value);
+        SuccessOrExit(error = ParseLong(aArgs[2], value));
+        VerifyOrExit((value == 0) || (value == 1), error = kErrorInvalidArgs);
+        level = static_cast<bool>(value);
```

Comment:
I think can use `ParseBool()` here instead?
```suggestion
        SuccessOrExit(error = ParseBool(aArgs[2], level));
```